src/components/GearItem.tsx

```diff
@@ -0,0 +1,99 @@
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Button from "@material-ui/core/Button";
+import ButtonGroup from "@material-ui/core/ButtonGroup";
+import React, { FunctionComponent } from "react";
+import ErrorIcon from "@material-ui/icons/Error";
+import MenuItem from "@material-ui/core/MenuItem";
+import Select from "@material-ui/core/Select";
+import Equipment from "../resources/Equipment";
+
+interface GearItemProps {
+  item: Equipment;
+  quantity: number;
+  setFieldValue: (field: string, value: number | string | boolean) => void;
+}
+const GearItem: FunctionComponent<GearItemProps> = ({
+  item,
+  quantity,
+  setFieldValue,
+}) => {
+  // Create all possible quantities for current item
+  const selectOptions: JSX.Element[] = [];
+  for (let i = 0; i <= item.quantity; ++i) {
+    selectOptions.push(<MenuItem value={i}>{i}</MenuItem>);
+  }
```

Comment:
Firstly, I think you actually have a bug here.  If your item looks like  `{quantity: 1}` then the loop will run for `i = 0` _and_ `i = 1`, so you get _two_ `<MenuItem>`... is that intended?

Secondly, this is definitely code that shouldn't be  in a component.  `for` loops are persona non grata here.  They are buggy and hard to read.  I would still like to extract it, but consider something like this (inside a well labelled function!) instead
```suggestion
  const selectOptions: JSX.Element[] = Array.from({length: item.quantity}).map((_, i) => <MenuItem value={i}>{i}</MenuItem>)
```